### PR TITLE
Add meals post endpoint

### PR DESCRIPTION
**Pull request by TheKaterTot**
This adds the post to /meals, allowing for foods to be added to a meal. It has a happy path test.

### CODE REVIEW

File: lib/controllers/meals-controller.js [hunk 1]
```diff
@@ -13,6 +13,13 @@ class MealsController {
       res.json(data.rows)
     })
   }
+
+  create(req, res) {
+    Meal.create(req.body.meal)
+    .then((data) => {
```
Comment (cmacaulay) [inline lib/controllers/meals-controller.js:19]: 🎉 ES6

File: lib/models/meal.js [hunk 2]
```diff
@@ -14,6 +14,23 @@ class Meal {
       [`${date.year}/${date.month}/${date.day}`, category])
   }
 
+  static create(mealAttrs) {
```
Comment (cmacaulay) [inline lib/models/meal.js:17]: whoa - what does `static` do @TheKaterTot?

File: lib/models/meal.js [hunk 3]
```diff
@@ -14,6 +14,23 @@ class Meal {
...
+                           [mealAttrs.date, categoryID, mealAttrs.foodID])
+    }).then((data) => {
+      return database.raw(`SELECT meals.*, categories.name AS category
+                           FROM meals
+                           JOIN categories ON category_id = categories.id
+                           WHERE meals.id = ?`,
+                           [data.rows[0].id])
+    })
+  }
+
```
Comment (cmacaulay) [inline lib/models/meal.js:33]: SQL 🌈 🌈 
- Makes sense! 
- Wizardry!

File: lib/routers/meals-router.js [hunk 4]
```diff
@@ -7,4 +7,6 @@ MealsRouter.get('/:category/:year/:month/:day', (req, res) => {
   mealsController.index(req, res);
 })
 
+MealsRouter.post('/', mealsController.create)
```
Comment (cmacaulay) [inline lib/routers/meals-router.js:10]: ✈️

File: test/meals-response-test.js [hunk 5]
```diff
@@ -228,5 +228,46 @@ describe('Server', () => {
         })
       })
     })
+
```
Comment (cmacaulay) [inline test/meals-response-test.js:231]: Testing! 👏 👏

File: test/meals-response-test.js [hunk 6]
```diff
@@ -228,5 +228,46 @@ describe('Server', () => {
...
+          assert.equal(parsedMeal.food_id, this.food.id)
+          assert.equal(parsedMeal.category, 'breakfast')
+          assert.equal(parsedDate.getFullYear(), '2017')
+          assert.equal(parsedDate.getMonth(), '4')
+          assert.equal(parsedDate.getDate(), '15')
+
+          done();
+        })
+      })
+    })
```
Comment (cmacaulay) [inline test/meals-response-test.js:271]: 🎉 🎉

**Review by cmacaulay — APPROVED** [review]
Great work!

File: lib/models/meal.js (hunk 2 above)
Comment (TheKaterTot) [reply lib/models/meal.js:17]: class method! I looked it up 👍

**Pull request merged**